src/usb/noise.c

```diff
@@ -214,7 +214,7 @@ void noise_rand_bytes(void* bytes, size_t size)
 // processes client messages. The first two messages are handshake messages
 // (see XX in https://noiseprotocol.org/noise.html#interactive-handshake-patterns-fundamental).
 // After, all incoming messages are decrypted and outgoing messages encrypted.
-bool bb_noise_process_msg(
+bb_noise_result_t bb_noise_process_msg(
```

Comment:
why was this change needed? seems like a 1:1 mapping, with `BB_NOISE_DEFER_REPLY` being unused.

Reply:
As per the commit description:
> Having the possibility of returning ok + don't send responses will allow to
defer the responses for later, thus moving the blocking calls outside of
the USB rx stack.

The idea is that this allows to start decoupling the need for responding immediately to all requests. If we want to make any workflow asynchronous in the future (so we can e.g. still detect disconnection from the app when we're in a blocking user operation and cancel it), we need to have a way to tell the USB stack that the operation is not completed yet, and so no response must be sent over USB (which is the third option for `bb_noise_result_t`). On a separate side, I think it should be possible to move this to a separate PR as it only affects the HWW stack.

Reply:
This has been moved to a separate branch and will come as a future PR.